src/mame/sony_news/cxd8442q.h

```diff
@@ -0,0 +1,146 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+
+/*
+ * Sony CXD8442Q WSC-FIFOQ APbus FIFO and DMA controller
+ *
+ * The FIFOQ is an APbus DMA controller designed for interfacing some of the simpler and lower speed peripherals
+ * to the APbus while providing DMA capabilities. Each FIFO chip can support up to 4 devices.
+ */
+
+#ifndef MAME_MACHINE_CXD8442Q_H
+#define MAME_MACHINE_CXD8442Q_H
+
+#pragma once
+
+class cxd8442q_device : public device_t
+{
+protected:
+	// Class that represents a single FIFO channel. Each instance of the FIFOQ chip has 4 of these.
+	class apfifo_channel
+	{
+	public:
+		apfifo_channel(cxd8442q_device &fifo_device) : fifo_device(fifo_device), dma_r_callback(fifo_device), dma_w_callback(fifo_device)
+		{
+		}
+
+		static constexpr int DMA_EN = 0x1;
+		static constexpr int DMA_DIRECTION = 0x2; // 1 = out, 0 = in
```

Comment:
Should these be `uint32_t` to match the value they’re tested against?

Reply:
Good catch, thanks!